app/src/componentLibrary/popover/withPopover/withPopover.jsx

```diff
@@ -26,40 +28,39 @@ export const withPopover = ({
   popoverWrapperClassName,
   tabIndex,
   ...popoverConfig
-}) => (
-  WrappedComponent,
-  // eslint-disable-next-line react/prop-types
-) => ({ isOpenPopover, closePopover, setIsOpenPopover, ...props }) => {
+}) => (WrappedComponent) => ({ isOpenPopover, togglePopover, ...props }) => {
   const parentRef = useRef();
   const [isOpened, setOpened] = useState(false);
 
-  useEffect(() => {
-    if (isOpenPopover) {
-      setOpened(true);
-      setIsOpenPopover?.(true);
+  const onClose = useCallback(() => {
+    if (togglePopover) {
+      togglePopover(false);
+    } else {
+      setOpened(false);
     }
-  }, [isOpenPopover, setIsOpenPopover]);
+  }, [togglePopover, setOpened]);
 
-  const onClose = () => {
-    closePopover?.();
-    setIsOpenPopover?.(false);
-    setOpened(false);
+  const onClickHandle = () => {
+    if (togglePopover) {
+      togglePopover((prev) => !prev);
```

Comment:
I'm not sure that this handler will always be state setter function, it can be just a callback that do some stuff internally.
So let's just call it here `togglePopover()` and leave the state changing logic in consumer's code.

Reply:
Done.